sdk/src/logs/logger_provider.cc

```diff
@@ -26,6 +27,7 @@ LoggerProvider::LoggerProvider(std::unique_ptr<LogRecordProcessor> &&processor,
   std::vector<std::unique_ptr<LogRecordProcessor>> processors;
   processors.emplace_back(std::move(processor));
   context_ = std::make_shared<sdk::logs::LoggerContext>(std::move(processors), std::move(resource));
+  OTEL_INTERNAL_LOG_DEBUG("[LoggerProvider] LoggerProvider created.");
```

Comment:
Do you mean to enhance our debug logging in general? That's good feedback, will create a separate issue for it, as it is not directly related to this fix.

Reply:
> How about other ctors? Didn't add the log there because they are not used by lazy singleton pattern? Probably add the log for consistency as the log message is provider instance created.

The per signal singletons are the entry points to the SDK. No other code in the SDK can be executed before building these per signal singletons.

So, by definition, once these singletons are executed, it is guaranteed that every other class used in the SDK implementation will see an already initialized global log handler.

There is not need, to fix this bug, to add log statements in every classes, even if another class uses a lazy singleton pattern.